app/Models/Plan/UserPlan.php

```diff
@@ -123,15 +124,45 @@ private function completeDaysCurrentUserPlan() : void
         }
     }
 
-    public function reset($start_date = null)
+    /**
+     * Reset the user plan according the given start date. If the save progress flag is true the
+     * plan progress will be removed.
+     *
+     * @param string $start_date
+     * @param bool   $save_progress
+     * @param int    $user_id
+     *
+     * @return Array
+     */
+    public function reset(string $start_date = null, bool $save_progress = false, ?int $user_id = null) : UserPlan
     {
-        PlanDay::where('plan_id', $this->plan_id)->get()
-            ->each(function ($plan_day) {
-                $plan_day->unComplete();
-            });
+        if ($save_progress) {
```

Comment:
This should be NOT save_progress. we only want to delete the stuff if the user does not want to save it. Otherwise, if they want to save progress, leave this alone